### PR TITLE
Improve pageUnloadBarrier waiting logic (closes #245)

### LINKED ISSUE #245: Implement new autowaits system
We need better heuristics for page unloads: currently we always wait at least a second after the action. We've discussed this in person, @AlexanderMoskovkin can you write down proposed algorithm here, please?

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/cc @inikulin @helen-dikareva

### CODE REVIEW

**Comment by inatakim** [comment]
@inikulin @helen

File: src/client/core/page-unload-barrier.js [hunk 1]
```diff
@@ -21,6 +21,9 @@ var waitingForUnloadTimeoutId = null;
 var waitingPromiseResolvers   = [];
 var unloading                 = false;
 
+var pageNavigationTriggeredListener = false;
```
Comment (helen-dikareva) [inline src/client/core/page-unload-barrier.js]: I think it should be `null`

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by inatakim** [comment]
696f368

**Comment by inatakim** [comment]
Tq

**Pull request merged**